### PR TITLE
Film to angular coordinates conversion test

### PR DESCRIPTION
**Pull request by nollety**
# Description

Resolves eradiate/eradiate-issues#122

Added: 15 tests (3 parametrised)
Total run time: 8 seconds

# Checklist

- [x] The code follows the relevant coding guidelines
- [x] The code generates no new warnings
- [x] The code is appropriately documented
- [x] The code is tested to prove its function
- [x] The feature branch is rebased on the current state of the `main` branch
- [x] I give permission that the Eradiate project may redistribute my contributions under the terms of its license

### CODE REVIEW

File: tests/_system/test_onedim_rpv.py [hunk 1]
```diff
@@ -0,0 +1,425 @@
...
+    Within the scope of this test, we define 'forward scattering' by
+    'the mean BRF value is larger in the side of the principal plane where the
+    illumination is not, than in the other side', and 'backward scattering'
+    by not forward scattering.
+
+    Rationale
+    ---------
+
+    * Geometry: a square surface with unit size and a RPV BRDF with
+      reflectance :math:`\rho \in [0.0, 0.3, 0.5, 0.7, 1.0]`.
```
Comment (leroyvn) [inline tests/_system/test_onedim_rpv.py:51]: I assume you mean `\rho_0` here?

File: tests/_system/test_onedim_lambertian_brf.py [hunk 2]
```diff
@@ -6,6 +6,7 @@
 import numpy as np
 
 import eradiate
+from eradiate import unit_registry as ureg
```
Comment (leroyvn) [inline tests/_system/test_onedim_lambertian_brf.py:9]: Do you really need to add this import here?

File: tests/_system/test_onedim_rpv.py [hunk 3]
```diff
@@ -0,0 +1,425 @@
...
+    surface.
+
+    We set a `distant_flux` measure to compute the sector radiosity at infinite
+    distance in the hemisphere whose direction is the upward direction.
+
+    Rationale
+    ---------
+
+    * Geometry: a square surface with unit size and a RPV BRDF with
+      reflectance :math:`\rho = 0.1` and with asimmetry parameter :math:`g` set
```
Comment (leroyvn) [inline tests/_system/test_onedim_rpv.py:176]: Ditto

**Review by leroyvn — COMMENTED** [review]
Hi @nollety, I know this PR is not ready for review yet but I had a few minor comments.

File: tests/_system/test_onedim_rpv.py (hunk 1 above)
Comment (nollety) [reply tests/_system/test_onedim_rpv.py]: Indeed, thanks!

File: tests/_system/test_onedim_lambertian_brf.py (hunk 2 above)
Comment (nollety) [reply tests/_system/test_onedim_lambertian_brf.py]: No I don't, thanks ^^

**Comment by nollety** [comment]
@leroyvn Thanks for your quick feedback, indeed this is still ongoing work ^^

**Comment by nollety** [comment]
Below are some graphs that show how the scattering behavior of the scene -- i.e. either foward-scattering or backward-scattering -- is evaluated. In the principal plane, the BRF in the 'forward' side of the plane is averaged and compared to the average of the BRF in the backward side of the plane. In the hemisphere, similarly to the BRF in the principal plane, the sector radiosity is averaged  in the region of the hemisphere corresponding to forward and backward scattering directions and compared.

test_film_to_angular_coord_conversion_multi_distant_forward_0 0

test_film_to_angular_coord_conversion_multi_distant_backward_0 0





test_film_to_angular_coord_conversion_distant_flux_forward_0 0

test_film_to_angular_coord_conversion_distant_flux_backward_0 0

**Comment by leroyvn** [comment]
Looks nice. Would you mind setting an aspect ratio of 1 for the image plots? (e.g. using [`Axes.set_aspect("equal")`](https://matplotlib.org/stable/api/_as_gen/matplotlib.axes.Axes.set_aspect.html))

**Comment by leroyvn** [comment]
Hi @nollety, looks good to me. I'll merge this as soon as you confirm that it's good to go!

**Comment by nollety** [comment]
Hi @leroyvn, this is good to go! 😃

**Pull request merged**